transaction/core/src/lib.rs

```diff
@@ -45,3 +56,26 @@ pub fn get_tx_out_shared_secret(
 ) -> RistrettoPublic {
     create_shared_secret(tx_public_key, view_key)
 }
+
+/// Helper which checks if a particular subaddress of an account key matches a
+/// TxOut
+///
+/// This is not the most efficient way to check when you have many subaddresses,
+/// for that you should create a table and use
+/// recover_public_subaddress_spend_key directly.
+///
+/// However some clients are only using one or two subaddresses.
+/// Validating that a TxOut is owned by the change subaddress is a frequently
+/// needed operation.
+pub fn subaddress_matches_tx_out(
```

Comment:
I'm adding this to try to make it easier for clients to figure out how to validate destination memos, it also makes some tests shorter